nifi-commons/nifi-expression-language/src/main/java/org/apache/nifi/attribute/expression/language/evaluation/functions/MathEvaluator.java

```diff
@@ -0,0 +1,138 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.attribute.expression.language.evaluation.functions;
+
+import org.apache.nifi.attribute.expression.language.evaluation.Evaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.NumberEvaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.NumberQueryResult;
+import org.apache.nifi.attribute.expression.language.evaluation.QueryResult;
+import org.apache.nifi.attribute.expression.language.exception.AttributeExpressionLanguageException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Map;
+
+public class MathEvaluator extends NumberEvaluator {
+
+    private final Evaluator<Number> subject;
+    private final Evaluator<String> methodName;
+    private final Evaluator<Number> optionalArg;
+
+    public MathEvaluator(final Evaluator<Number> subject, final Evaluator<String> methodName, final Evaluator<Number> optionalArg) {
+        this.subject = subject;
+        this.methodName = methodName;
+        this.optionalArg = optionalArg;
+    }
+
+    @Override
+    public QueryResult<Number> evaluate(final Map<String, String> attributes) {
+        final String methodNamedValue = methodName.evaluate(attributes).getValue();
+        if (methodNamedValue == null) {
+            return new NumberQueryResult(null);
+        }
+
+        final Number subjectValue;
+        if(subject != null) {
+            subjectValue = subject.evaluate(attributes).getValue();
+            if(subjectValue == null){
+                return new NumberQueryResult(null);
+            }
+        } else {
+            subjectValue = null;
+        }
+
+        final Number optionalArgValue;
+        if(optionalArg != null) {
+            optionalArgValue = optionalArg.evaluate(attributes).getValue();
+
+            if(optionalArgValue == null) {
+                return new NumberQueryResult(null);
+            }
+        } else {
+            optionalArgValue = null;
+        }
+
+        try {
+            Number executionValue = null;
+
+            if (subjectValue == null){
+                Method method = Math.class.getMethod(methodNamedValue);
+
+                if(method == null) {
+                    throw new AttributeExpressionLanguageException("Cannot evaluate 'math' function because no subjectless method was found with the name:'" + methodNamedValue + "'");
+                }
+
+                executionValue = (Number) method.invoke(null);
+            } else if(optionalArg == null) {
+                boolean subjectIsDecimal = subjectValue instanceof Double;
+                Method method = Math.class.getMethod(methodNamedValue, subjectIsDecimal ? double.class : long.class);
```

Comment:
You are totally right. I even hit that exception before and didn't think anything of it. I will re-work it a bit in order to hit the proper error message.